api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxRttCohortQueries.java

```diff
@@ -261,4 +263,143 @@ private CohortDefinition getPatientsWithFilaOrFichaOrMasterCardPickup(
+            + "                                AND        o1.value_datetime IS NOT NULL  "
+            + "                                AND        o1.voided=0  "
+            + "                                AND        o1.concept_id IN (${returnVisitDateConcept})  "
+            + "                                AND        o1.location_id = :location  "
+            + "                                UNION  "
+            + "                                SELECT     qc.patient_id  "
+            + "                                FROM       (  "
+            + "                                                      SELECT     pat.patient_id,  "
+            + "                                                                 max(e.encounter_datetime) AS encounter_datetime "
+            + "                                                      FROM       patient pat  "
+            + "                                                      INNER JOIN encounter e  "
+            + "                                                      ON         pat.patient_id=e.patient_id  "
+            + "                                                      WHERE      e.encounter_datetime<=:onOrBefore "
+            + "                                                      AND        pat.voided=0  "
+            + "                                                      AND        e.voided=0  "
+            + "                                                      AND        e.location_id=:location  "
+            + "                                                      AND        e.encounter_type IN(${aRVPharmaciaEncounterType}) "
+            + "                                                      GROUP BY   pat.patient_id)qc  "
+            + "                                INNER JOIN encounter e1  "
+            + "                                ON         qc.patient_id=e1.patient_id  "
+            + "                                INNER JOIN obs o1  "
+            + "                                ON         e1.encounter_id=o1.encounter_id  "
+            + "                                WHERE      qc.encounter_datetime=e1.encounter_datetime  "
+            + "                                AND        e1.encounter_datetime<=:onOrBefore  "
+            + "                                AND        e1.voided=0  "
+            + "                                AND        e1.encounter_type IN(${aRVPharmaciaEncounterType})  "
+            + "                                AND        e1.location_id=:location  "
+            + "                                AND        o1.value_datetime IS NOT NULL  "
+            + "                                AND        o1.voided=0  "
+            + "                                AND        o1.concept_id IN(${returnVisitDateForArvDrugConcept})  "
+            + "                                AND        o1.location_id = :location  "
```

Comment:
I think it's missing the without of this specs: 

> the most recent Encounters of Type 6, 9 and 18 by reporting endDate without the following observations or its contents is null:
Next Clinical Appointment (Concept ID 1410)
Next Drugs Pick Up Appointment (Concept ID 5096)